nodes/test/nodes_tests/test_plugins.py

```diff
@@ -186,7 +190,7 @@ def tearDown(self):
         shutil.rmtree(self.childfs)
         Consumer.get_collection().remove()
         Bind.get_collection().remove()
-        Repo.get_collection().remove()
+        model.Repository.drop_collection()
```

Comment:
On 06/18/2015 09:29 AM, Barnaby Court wrote:
> not sure it makes a big difference, remove clears the contents of a
> collection but the collection remains. drop_collection deletes the
> collection as well as the contents. I noticed the change in behavior and
> thus was curious as to why.

There is one way in which it makes an important difference. As said,
remove() removes the documents in the collection and drop_collection()
drops the collection itself. The significant bit is that the indices get
dropped as well with drop_collection() but they remain with remove().

Since we hopefully aren't writing large amounts of data in our unit
tests, it probably doesn't make a noticeable difference in test
performance if the indices are present or not. But technically it is
probably slightly advantageous for write-once read-many tests (if we
have such things).

This actually could affect correctness too. We do have some code that
relies on uniqueness constraints. If the tests drop the database
enforced uniqueness constraints, the code being tested isn't going to
benefit from the more realistic environment that the code would
experience in production. That said, I think our tests should move away
from using the database entirely so perhaps this point is moot.

-- 
Randy Barlow